scopes/component/snapping/version-maker.ts

```diff
@@ -498,17 +495,43 @@ export class VersionMaker {
         }
       });
     };
+    const componentRangePrefix = this.dependencyResolver.componentRangePrefix();
+    const updateDepsResolverData = (component: ConsumerComponent) => {
+      const entry = component.extensions.findCoreExtension(DependencyResolverAspect.id);
+      if (!entry) {
+        return component;
+      }
+      const dependencies = get(entry, ['data', 'dependencies'], []);
+      dependencies.forEach((dep) => {
+        if (dep.__type !== COMPONENT_DEP_TYPE) {
+          return;
+        }
+        // @todo: it's unclear why "dep.componentId" randomly becomes a ComponentID instance.
+        // this check is added because on Ripple in some scenarios it was throwing:
+        // "ComponentID.fromObject expect to get an object, got an instance of ComponentID" (locally it didn't happen)
+        const depId =
+          dep.componentId instanceof ComponentID ? dep.componentId : ComponentID.fromObject(dep.componentId);
```

Comment:
better to not check instance of here as you might get different instance maybe I think